scanner/indexer/indexer.go

```diff
@@ -363,6 +368,13 @@ func getLayerRequest(ctx context.Context, httpClient *http.Client, imgRef name.R
 	}
 	utils.IgnoreError(res.Body.Close)
 	if res.StatusCode != http.StatusPartialContent {
+		if _, exists := regsNoRange.Load(registryURL.Host); !exists {
+			regsNoRange.Store(registryURL.Host, struct{}{})
```

Comment:
Sending requests to Scanner with made-up registries returning `StatusPartialContent` becomes a door for a DooS with this.

An LRU cache instead of an ever-growing map would solve this.

Reply:
makes sense. done in the followup commit, thanks